pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -616,9 +616,50 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		changes.KillPod = true
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+		if pod.Spec.RestartPolicy != v1.RestartPolicyAlways && changes.KillPod == false {
+			onlySidecars := true
+
+			//Determine if there are any non sidecar containers that are running or need restarting
+			//if there are none, we can kill the remaining sidecars
+			for _, container := range pod.Spec.Containers {
+				if !lifecycleSidecar(container) {
+					containerStatus := podStatus.FindContainerStatusByName(container.Name)
+					if kubecontainer.ShouldContainerBeRestarted(&container, pod, podStatus) || containerStatus.State == kubecontainer.ContainerStateRunning {
+						onlySidecars = false
+					}
+				}
+			}
+
+			//only sidecars are left so terminate them all
+			if onlySidecars {
```

Comment:
mark kill pod == true as well since we are killing all sidecars?

Reply:
yeah fixed.